### PR TITLE
tpm2_auth_util: cleanup session auth value parsing

### PR DESCRIPTION
**Pull request by williamcroberts**
Rather than have the general entry point deal with the parsing
pf the +, which would break passwords with a + in them, check
if it is a session, copy and modify the argument spliting the
string on the + character and call into the password handler for
the remaining password string if present.

This will support passwords with strings, and remove some
complexity from the code (confusing loop).

Signed-off-by: William Roberts <william.c.roberts@intel.com>

### CODE REVIEW

File: lib/tpm2_auth_util.c [hunk 1]
```diff
@@ -110,48 +87,77 @@ static bool handle_str(const char *password, TPMS_AUTH_COMMAND *auth) {
...
+        return false;
+    }
 
-        if (result) {
-            continue;
-        } else {
-            free((char*)password_string_ptr);
-            return false;
+    /*
+     * Sessions can have an associated password/auth value denoted adter
```
Comment (idesai) [inline lib/tpm2_auth_util.c]: typo -- after

File: lib/tpm2_auth_util.c [hunk 2]
```diff
@@ -110,48 +87,77 @@ static bool handle_str(const char *password, TPMS_AUTH_COMMAND *auth) {
...
+     * a + sign, ie session.ctx+foo, deal with it by
```
Comment (idesai) [inline lib/tpm2_auth_util.c]: is the order restricted here. As in can one do hex:password+session:session.ctx?

Comment (williamcroberts) [reply lib/tpm2_auth_util.c]: No that wouldn't work. the flow is session->password as in the session handler we check if their is a +, and if it contains an entry, call the handle_password routine which ONLY handles passwords.

Comment (idesai) [reply lib/tpm2_auth_util.c]: Cool. Can you please add this info to https://github.com/tpm2-software/tpm2-tools/blob/master/man/common/authorizations.md?

File: man/common/authorizations.md [hunk 3]
```diff
@@ -29,4 +29,23 @@ HMAC tickets can be presented as hex escaped passwords.
 
 When using a policy session to authorize the use of an object, prefix the option argument
 with the *session* keyword.  Then indicate a path to a session file that was created
-with tpm2_startauthsession(1).
+with tpm2_startauthsession(1). Optionally, if the session requires an auth value to be
+sent with the session handle (eg policy secret), then append a + and a string as described
```
Comment (idesai) [inline man/common/authorizations.md]: eg policypassword

Comment (williamcroberts) [reply man/common/authorizations.md]: Darn.. always switch those two!

Comment (idesai) [reply man/common/authorizations.md]: @williamcroberts this PR is definitely much better way of handling this

**Pull request merged**